### PR TITLE
perf: lazy validators (ME-15925)

### PR DESCRIPTION
**Pull request by tsi**
This PR adds a debug mode that will determine the appearance of dev logs and warnings

### CODE REVIEW

File: src/plugins/videojs-http-source-selector/components/SourceMenuButton.js [hunk 1]
```diff
@@ -36,8 +36,7 @@ class SourceMenuButton extends MenuButton {
     }
 
     // Bind update to qualityLevels changes
-    // Todo: switch to Function.prototype.bind
-    this.player_.qualityLevels().on(['change', 'addqualitylevel', 'removequalitylevel'], videojs.bind(this, this.update));
+    this.player_.qualityLevels().on(['change', 'addqualitylevel', 'removequalitylevel'], this.update.bind(this));
```
Comment (tsi) [inline src/plugins/videojs-http-source-selector/components/SourceMenuButton.js:39]: Suppress a deprecation console warning regarding `videojs.bind`

File: src/video-player.js [hunk 2]
```diff
@@ -72,14 +71,15 @@ class VideoPlayer extends Utils.mixin(Eventable) {
...
-
-    isValidConfig(this.options, playerValidators);
-
-    if (!this._isPlayerConfigValid) {
-      this.videojs.error('invalid player configuration');
-      return;
+    if (this.playerOptions.debug) {
+      isValidPlayerConfig(this.options).then((valid) => {
+        if (!valid) {
+          this._isPlayerConfigValid = valid;
```
Comment (ehab-cl) [inline src/video-player.js:78]: why not set it directly to false?

File: src/video-player.js [hunk 3]
```diff
@@ -390,7 +390,7 @@ class VideoPlayer extends Utils.mixin(Eventable) {
   _initAnalytics() {
     const analyticsOpts = this.playerOptions.analytics;
 
-    if (!window.ga && analyticsOpts) {
+    if (!window.ga && analyticsOpts && this.playerOptions.debug) {
```
Comment (ehab-cl) [inline src/video-player.js:393]: For later i think it is better to create an abstract that wraps the console.error statement and considers the debug mode.

**Review by ehab-cl — APPROVED** [review]

File: src/plugins/videojs-http-source-selector/components/SourceMenuButton.js (hunk 1 above)
Comment (tsi) [reply src/plugins/videojs-http-source-selector/components/SourceMenuButton.js:39]: I don't think so, but let's wait with it till next release

**Pull request merged**